src/Maker/MakeConvertPhpServices.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Maker;
+
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\Exception\RuntimeCommandException;
+use Symfony\Bundle\MakerBundle\FileManager;
+use Symfony\Bundle\MakerBundle\Generator;
+use Symfony\Bundle\MakerBundle\InputConfiguration;
+use Symfony\Bundle\MakerBundle\Util\PhpServicesCreator;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+
+/**
+ * @author Antoine Michelet <jean.marcel.michelet@gmail.com>
+ */
+final class MakeConvertPhpServices extends AbstractMaker
+{
+    private $fileManager;
+
+    public function __construct(FileManager $fileManager)
+    {
+        $this->fileManager = $fileManager;
+    }
+
+    public static function getCommandName(): string
+    {
+        return 'make:convert-php-services';
+    }
+
+    public function configureDependencies(DependencyBuilder $dependencies)
+    {
+    }
+
+    public function configureCommand(Command $command, InputConfiguration $inputConf)
+    {
+        $command->setDescription('Converts services.yaml file to services.php');
+
+        $command->addArgument('warning-comment',
+            InputArgument::REQUIRED,
+            'This command will completely remove your services.yaml file after completion. Ready to convert to services.php? Y/n'
+        );
+    }
+
+    public function generate(InputInterface $input, ConsoleStyle $io, Generator $generator)
+    {
+        if ('n' === strtolower($input->getArgument('warning-comment'))) {
+            return;
+        }
+
+        if (!$this->fileManager->fileExists($path = 'config/services.yaml')) {
+            throw new RuntimeCommandException('The file "config/services.yaml" does not exist.');
+        }
+
+        $phpServices = (new PhpServicesCreator())->convert(
+            $this->fileManager->getFileContents('config/services.yaml')
+        );
+
+        $this->fileManager->dumpFile('config/services.php', $phpServices);
+
+        $this->fileManager->removeFile('config/services.yaml');
```

Comment:
Sorry, I didn't quite say the correct thing in my original review. `$this->fileManager->fileExists()` and `$this->fileManager->getFileContents()` is correct. However, both `$this->fileManager->dumpFile()` and `$this->fileManager->removeFile()` should use the `$generator` variable. This means that you will need to add a new `removeFile()` method to `Generator`. That would probably save things in some new `$pendingFileDeletions` private property in `Generator`. Then, in `Generator:: writeChanges()`, we would loop over `$this->pendingFileDeletions` and call the new `removeFile` method that you added to `FileManager`.

Why? Well, I realize that we're a bit inconsistent inside MakerBundle itself, but you're *supposed* to use `Generator` for all file operations, like updating or deleting files. It helps all the file changes print out at the bottom of the command, instead of randomly while the command is running.

Reply:
Done.

Reply:
I forgot redo this. i am working on now.

Reply:
Okay, done